### ISSUE TITLE
Pragram crashed when joycon connected to mac with the first release version

### ISSUE THREAD

**Issue by forrany** [marcowindt/ds4macos#2]
HI, I update my mac os to 11.2.3 (20D91), just to fulfil the requirements of first release version.

But unfortunately,  the pragram will crash as long as my joycon connected to mac by bluetooth, any idea to solve this problem? Thanks

![image](https://user-images.githubusercontent.com/29779326/111858186-ee76b980-8971-11eb-825a-79144cbac2c2.png)

Image Translation: 

"ds4macos" Quit unexpectedly
Tap "Reopen" to open the app again. Tap "Report" to view more detailed information and send the report to Apple.

reopen
report...
ignore

**Comment by Microflamer** [comment]
I have this issue to. It would be nice if this was fixed as I really want to use joycons with Dolphin. There is some demand for it as well:https://www.reddit.com/r/DolphinEmulator/comments/kbgohx/is_it_possible_to_use_joycons_on_mac_and_use/

https://www.reddit.com/r/DolphinEmulator/comments/jr4o71/help_how_can_i_connect_joycons_to_dolphin_mac/

**Comment by marcowindt** [comment]
Sorry for the late response!

The [mac OS GameController Framework](https://developer.apple.com/documentation/gamecontroller) does support the JoyCon controller buttons, but it will not allow you to read the motion input (gyroscope/accelerometer). The program crashed in attempt on reading the motion sensor data from the device. Unfortunately there is no fix for this at the moment to get motion sensor data using the GameController Framework. (Latest commit https://github.com/marcowindt/ds4macos/commit/6313f64f5fb0154c95a7707b68731f6d7626f4cf fixes the program crashing, but no sensor data)

However, I found a library on github [JoyConSwift](https://github.com/magicien/JoyConSwift) which allows you to read the motion sensor data without using the GameController Framework. I created a simple app [joy4macos](https://github.com/marcowindt/joy4macos) which uses this library to send motion data to Dolphin, but it's still very buggy and I don't actually own any JoyCon controllers myself to run more tests. I will run some more tests when I get my hands on one again.
